_projects/automate_vulture_whitelist.md

```diff
@@ -0,0 +1,86 @@
+---
+name: "Automate Whitelist creation for vulture"
+collaborating_projects:
+  - coala
+  - vulture
+desc: "Automatically create whitelists and detect false positives for vulture."
+developers_involved:
+  - RJ722
+mentors:
+  - AbdealiJK
+  - jendrikseipp
+  - The-Compiler
+difficulty: medium
+status: in_progress
+initiatives:
+  - GSoC
+issues: []
+markdown: automate_vulture_whitelist.md
+requirements:
+  - "Familiarity with vulture and how whitelists work."
+tags:
+  - Vulture
+  - AST
+  - coverage
+---
+
+The main aim of this project is to automatically create whitelists (to be
+utilized by vulture) for projects projects using
+[sip](https://www.riverbankcomputing.com/software/sip/intro) and
+[`shiboken`](http://code.qt.io/cgit/pyside/pyside-setup.git/tree/sources/shiboken2)
+to generate Python bindings for C++ code, most significant of them being PyQt.
+Once implemented, a whitelist generated for PyQt (let's say) through this
+generator could then be universally utilised across all PyQt projects.
+
+#### IDEAS
+
+* Create a command line tool which takes in sip files as input and spits a
+  whitelist with all the virtual functions defined in the sip files. (Add
+  Documentation and tests)
+* Use this tool to generate a whitelist for PyQt and ship it with vulture. This
+  will reduce overhead for PyQt projects which use vulture.
+* Allow passing coverage.xml to vulture via `--make-whitelist` and let it find
+  the list of false-positive functions. (Add Documentation and tests as well)
+* Write a whitelist generator for shiboken, the binding
+  generator used for PySide2
+
+#### MILESTONES
+
+##### COMMUNITY BONDING
+* Discussion on whether to implement a new Tracer or to directly consume `xml`
+  output from coverage.py
+* Get familiar with coverage.xml
+* Take up small tasks from
+  [TODO](https://github.com/jendrikseipp/vulture/tree/master/TODO.rst)
+* [Follow up milestone](https://gitlab.com/coala/GSoC/GSoC-2018/milestones/28)
+
+##### CODING PHASE 1
+* The whitelist generator should be completely implemented.
+* It should reside in it's own repository, with CI enabled for testing on both
+  the GNU/Linux and Windows platforms. CI should also ensure ongoing support
+  for 2.7 & py3.4+ for compatibility with vulture.
+* Write tests, achieve 100% code coverage.
+* It should be gracefully documented.
+* A whitelist for PyQt should be generated to be then shipped with vulture by
+  default, thus helping PyQt projects (like
+  [qutebrowser](https://github.com/qutebrowser/qutebrowser)) to deal with false
+  positives.
+* Maximise code coverage of vulture.
+* [Follow up milestone](https://gitlab.com/coala/GSoC/GSoC-2018/milestones/29)
```

Comment:
please do not include this type of bullet point about milestones in https://gitlab.com/coala/GSoC/GSoC-2018/

The project page should have been published before https://gitlab.com/coala/GSoC/GSoC-2018/ was created, and before the gsoc slots were selected.  You need to pretend those milestones dont exist yet.

If there is anything in those milestones which should be added to the project, they need to be added as bullet points here.